### PR TITLE
[diem-framework] Add support for unit testing in the Diem Framework

### PR DESCRIPTION
**Pull request by tzakian**
This PR adds support for unit testing in the Diem Framework, and ports over/adds unit tests for the `AccountFreezing` module. 

The changes are generally pretty minor, the only significant changes being:
* A small refactor of the genesis code so we don't pass ownership of the signers to it so it can be called easily in unit tests.
* Adding the ability to pass the set of native functions to the unit testing framework (needed to support DF-specific native functions).

### CODE REVIEW

**Review by tnowacki — DISMISSED** [review]
Looks good to me! 

@vgao1996 might want to look?

File: language/diem-framework/tests/move_unit_test.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_vm::natives::diem_natives;
+use move_unit_test::UnitTestingConfig;
+
+move_unit_test::register_move_unit_tests!(
+    UnitTestingConfig::default_with_bound(Some(100_000)),
+    ".",
+    r".*\.move$",
+    r"../move-stdlib/modules",
```
Comment (vgao1996) [inline language/diem-framework/tests/move_unit_test.rs:11]: Please avoid referencing contents in another crate via paths. This makes it hard for us to see the dependency.

I'd suggest you use `move_stdlib::move_stdlib_modules_full_path()` instead.

**Review by vgao1996 — CHANGES_REQUESTED** [review]
LGTM except for a path referencing a directory in move-stdlib.

One more question: is my understanding correct that the unit test framework does not have a way to invoke `Genesis::initialize_internal` automatically, and it has to be explicitly called by tests that need it?

File: language/diem-framework/tests/move_unit_test.rs (hunk 1 above)
Comment (tzakian) [reply language/diem-framework/tests/move_unit_test.rs:11]: Correct about genesis. We may want to add this functionality later (custom init functions), but I feel as though adding such a feature would run counter to the philosophy of the testing framework (minimality + transparency of its semantics).

**Review by vgao1996 — APPROVED** [review]

File: language/diem-framework/tests/move_unit_test.rs (hunk 1 above)
Comment (vgao1996) [reply language/diem-framework/tests/move_unit_test.rs]: Yeah the current solution actually looks pretty neat to me. I especially like the fact that it doesn't require a special mechanism for initialization.

**Pull request merged**